src/Umbraco.Core/Composing/Composers.cs

```diff
@@ -18,19 +18,30 @@ public class Composers
         private readonly Composition _composition;
         private readonly IProfilingLogger _logger;
         private readonly IEnumerable<Type> _composerTypes;
+        private readonly IEnumerable<Attribute> _enableDisableAttributes;
 
         private const int LogThresholdMilliseconds = 100;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="Composers"/> class.
+        /// Initializes a new instance of the <see cref="Composers" /> class.
         /// </summary>
         /// <param name="composition">The composition.</param>
-        /// <param name="composerTypes">The composer types.</param>
-        /// <param name="logger">A profiling logger.</param>
-        public Composers(Composition composition, IEnumerable<Type> composerTypes, IProfilingLogger logger)
+        /// <param name="composerTypes">The <see cref="IComposer" /> types.</param>
+        /// <param name="enableDisableAttributes">The <see cref="EnableComposerAttribute" /> and/or <see cref="DisableComposerAttribute" /> attributes.</param>
+        /// <param name="logger">The profiling logger.</param>
+        /// <exception cref="ArgumentNullException">composition
+        /// or
+        /// composerTypes
+        /// or
+        /// enableDisableAttributes
+        /// or
+        /// logger</exception>
+
+        public Composers(Composition composition, IEnumerable<Type> composerTypes, IEnumerable<Attribute> enableDisableAttributes, IProfilingLogger logger)
```

Comment:
@ronaldbarendse can we make this non-breaking - so that we obsolete the old constructor and also if the old constructor is used it fallbacks to the old way of finding the attributes?

Reply:
@ronaldbarendse what do you think abuot this breaking change?

Reply:
I've added the obsolete constructor for backwards compatibility, as both `CoreRuntime` and `Composers` classes are public and the `Boot` method can be overridden (and indeed still call this constructor overload) 👍